### PR TITLE
Apply liqp filters to template instead of globally

### PR DESCRIPTION
**Pull request by kormide**
@jbedard

### CODE REVIEW

File: recipe-generator/src/main/java/ca/derekcormier/recipe/generator/JavaGenerator.java [hunk 1]
```diff
@@ -11,13 +14,21 @@
 public abstract class JavaGenerator extends Generator {
     public JavaGenerator(Cookbook cookbook) {
         super(cookbook);
+    }
+
+    @Override
+    protected List<Filter> getTemplateFilters() {
+        List<Filter> filters = new ArrayList<>();
```
Comment (jbedard) [inline recipe-generator/src/main/java/ca/derekcormier/recipe/generator/JavaGenerator.java]: imo, when there's a hardcoded list of values like this, it's nicer to do `return Arrays.asList(....)`

File: recipe-generator/src/main/java/ca/derekcormier/recipe/generator/Generator.java [hunk 2]
```diff
@@ -33,6 +34,11 @@ protected String renderTemplate(String templatePath, Map<String,Object> data) {
         Scanner scanner = new Scanner(inputStream).useDelimiter("\\A");
         String templateContent = scanner.hasNext() ? scanner.next() : "";
         Template template = Template.parse(templateContent).withRenderSettings(new RenderSettings.Builder().withStrictVariables(false).build());
+
+        for (Filter filter: getTemplateFilters()) {
```
Comment (jbedard) [inline recipe-generator/src/main/java/ca/derekcormier/recipe/generator/Generator.java]: No space before the `:` 😲

File: recipe-generator/src/main/java/ca/derekcormier/recipe/generator/Generator.java [hunk 3]
```diff
@@ -41,6 +47,10 @@ protected String renderTemplate(String templatePath, Map<String,Object> data) {
         }
     }
 
+    protected List<Filter> getTemplateFilters() {
+        return new ArrayList<>();
```
Comment (jbedard) [inline recipe-generator/src/main/java/ca/derekcormier/recipe/generator/Generator.java]: `Collections.emptyList()`? 🤷‍♂️

**Review by jbedard — APPROVED** [review]
This API makes more sense to me 👍

File: recipe-generator/src/main/java/ca/derekcormier/recipe/generator/Generator.java (hunk 2 above)
Comment (kormide) [reply recipe-generator/src/main/java/ca/derekcormier/recipe/generator/Generator.java]: Never!

**Comment by kormide** [comment]
Updated

**Pull request merged**